### PR TITLE
Convert to akka 2.5

### PR DESCRIPTION
**Pull request by vjkoskela**
@BrandonArp 

Could you take a look at converting the flow code?

```
final UniformFanOutShape<HttpRequest, HttpRequest> split = builder.add(Broadcast.create(2));

                final FlowShape<HttpRequest, byte[]> getBody = builder.add(getBodyFlow);
                final FlowShape<HttpRequest, Multimap<String, String>> getHeaders = builder.add(getHeadersFlow);
                final FanInShape2<
                        byte[],
                        Multimap<String, String>,
                        Pair<byte[], Multimap<String, String>>> join = builder.add(Zip.create());
                final FlowShape<Pair<byte[], Multimap<String, String>>, Record> createRequest =
                        builder.add(createAndParseFlow);
```

### CODE REVIEW

**Comment by BrandonArp** [comment]
The Flow code didn't need to change.  I updated the akka HTTP deps and fixed some checkstyle and deprecation warnings.  Just need your sign off.

File: pom.xml [hunk 1]
```diff
@@ -195,6 +195,30 @@
...
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>analyze</id>
+            <goals>
+              <goal>analyze-only</goal>
+            </goals>
+            <configuration>
+              <failOnWarning>true</failOnWarning>
+              <!-- ignore annotations for "used but undeclared" warnings -->
```
Comment (vjkoskela) [inline pom.xml]: This is a bit concerning, if we've used it, why not just declare a dependency on it?

File: pom.xml [hunk 2]
```diff
@@ -195,6 +195,30 @@
...
+              <ignoredUsedUndeclaredDependencies>
+                <ignoredUsedUndeclaredDependency>com.typesafe.akka:akka-http-core_${scala.version}</ignoredUsedUndeclaredDependency>
+              </ignoredUsedUndeclaredDependencies>
+
+              <!-- ignore annotations for "unused but declared" warnings -->
+              <ignoredUnusedDeclaredDependencies>
+                <ignoredUnusedDeclaredDependency>com.typesafe.akka:akka-http_${scala.version}</ignoredUnusedDeclaredDependency>
```
Comment (vjkoskela) [inline pom.xml]: Should we just mark this as a runtime dep?

**Comment by vjkoskela** [comment]
Two questions. Otherwise ship it.

**Pull request merged**